common/src/chain/transaction/output/mod.rs

```diff
@@ -41,71 +41,64 @@ pub enum Destination {
     ClassicMultisig(PublicKeyHash),
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Encode, Decode)]
-pub enum OutputPurpose {
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+pub enum TxOutput {
     #[codec(index = 0)]
-    Transfer(Destination),
+    Transfer(OutputValue, Destination),
     #[codec(index = 1)]
-    LockThenTransfer(Destination, OutputTimeLock),
+    LockThenTransfer(OutputValue, Destination, OutputTimeLock),
     #[codec(index = 2)]
-    Burn,
+    Burn(OutputValue),
     /// Output type that is used to create a stake pool
     #[codec(index = 3)]
     StakePool(Box<StakePoolData>),
     /// Output type that represents spending of a stake pool output in a block reward
     /// in order to produce a block
     #[codec(index = 4)]
-    ProduceBlockFromStake(Destination, PoolId),
+    ProduceBlockFromStake(Amount, Destination, PoolId),
 }
 
-impl OutputPurpose {
+impl TxOutput {
+    // TODO: this has to go
```

Comment:
could you explain why?

Reply:
Because it makes assumptions about destination. A destination (and whether it exists, and whether there's more than one) depends on the type of the output. The correct way it to match this when needed, not make assumptions about what destination means, which can easily lead to misinterpretations.